src/messages/gateway.rs

```diff
@@ -62,6 +62,7 @@ pub enum FromGatewayMessage {
     CancelTasks(CancelTasks),
     GetTaskInfo(TaskInfoRequest),
     ServerInfo,
+    GetWorkerEvents(EventRequest),
```

Comment:
Let's make this a bit more general and create a new name for these events. I would suggest `MonitoringEvent`, so `GetMonitoringEvents`.